### PR TITLE
TT-1154 Add new OCR soak tests with gas simulations

### PR DESCRIPTION
**Pull request by lukaszcl**
This PR adds new gas simulation tests for OCR:
- `TestOCRSoak_GasSpike` test modulates the base fee per gas over a duration, reaching up to 3400000000 base fee. 
- `TestOCRSoak_ChangeBlockGasLimit` changes next block gas limit and sets it to percentage of last gasUsed in previous block creating congestion
 
The screenshots below are from the [example testrun](https://grafana.ops.prod.cldev.sh/goto/pdQOWE8Sg?orgId=1). The Base Fee panel shows how the base fee changes over time for a core node.

Screenshot 2024-06-17 at 17 06 50

image

image

### CODE REVIEW

**Review by skudasov — DISMISSED** [review]

**Review by skudasov — APPROVED** [review]

File: integration-tests/testsetups/ocr.go [hunk 1]
```diff
@@ -598,7 +601,47 @@ func (o *OCRSoakTest) testLoop(testDuration time.Duration, newValue int) {
 		} else {
 			reorgDelay = n.GethReorgConfig.DelayCreate.Duration
 		}
-		reorgCh = time.After(reorgDelay)
+		time.AfterFunc(reorgDelay, func() {
+			if !o.reorgHappened {
+				o.startGethBlockchainReorg(o.rpcNetwork, n.GethReorgConfig)
```
Comment (Tofel) [inline integration-tests/testsetups/ocr.go:606]: curious: why `GethBlockchainReorg` and not `Anvil...`?

File: integration-tests/testsetups/ocr.go [hunk 2]
```diff
@@ -598,7 +601,47 @@ func (o *OCRSoakTest) testLoop(testDuration time.Duration, newValue int) {
...
+			}
+		})
+	}
+
+	// Schedule gas simulations if needed
+	// Gas simulation only available for Anvil
+	if o.rpcNetwork.Name == "Anvil" {
+		ac := o.Config.GetNetworkConfig().AnvilConfigs["ANVIL"]
+		if ac != nil && ac.GasSpikeSimulation.Enabled {
```
Comment (Tofel) [inline integration-tests/testsetups/ocr.go:615]: can `GasSpikeSimulation` ever be `nil`?

File: integration-tests/testsetups/ocr.go [hunk 3]
```diff
@@ -655,22 +692,54 @@ func (o *OCRSoakTest) complete() {
...
+		Str("URL", client.URL).
+		Any("GasLimitSimulationConfig", conf).
+		Uint64("LatestBlock", latestBlock.Number().Uint64()).
+		Uint64("LatestGasUsed", latestBlock.GasUsed()).
+		Uint64("LatestGasLimit", latestBlock.GasLimit()).
+		Int64("NewGasLimit", newGasLimit).
+		Msg("Starting new gas limit simulation on Anvil chain")
+	err = client.AnvilSetBlockGasLimit([]interface{}{newGasLimit})
+	require.NoError(o.t, err, "Error starting gas simulation on Anvil chain")
+	time.Sleep(conf.Duration.Duration)
```
Comment (Tofel) [inline integration-tests/testsetups/ocr.go:732]: curious, why we need to sleep here?

File: integration-tests/testsetups/ocr.go (hunk 1 above)
Comment (lukaszcl) [reply integration-tests/testsetups/ocr.go:606]: @Tofel we use geth blockchain for rerorg scenarios by default

File: integration-tests/testsetups/ocr.go (hunk 2 above)
Comment (lukaszcl) [reply integration-tests/testsetups/ocr.go:615]: @Tofel no

File: integration-tests/testsetups/ocr.go (hunk 3 above)
Comment (lukaszcl) [reply integration-tests/testsetups/ocr.go:732]: @Tofel this is to wait before returning to the old gas limit in the next lines

**Review by Tofel — APPROVED** [review]

**Pull request merged**